starport/pkg/chaincmd/chaincmd.go

```diff
@@ -0,0 +1,252 @@
+	for _, applyOption := range options {
+		applyOption(&chainCmd)
+	}
+
+	return chainCmd
+}
+
+type Option func(*ChainCmd)
+
+// WithHome replaces the default home used by the chain
+func WithHome(home string) Option {
+	return func(c *ChainCmd) {
+		c.homeDir = home
+	}
+}
+
+// WithChainID provides a specific chain ID for the commands that accept this option
+func WithChainID(chainID string) Option {
+	return func(c *ChainCmd) {
+		c.chainID = chainID
+	}
+}
+
+// WithKeyrinBackend provides a specific keyring backend for the commands that accept this option
+func WithKeyrinBackend(keyringBackend KeyringBackend) Option {
+	return func(c *ChainCmd) {
+		c.keyringBackend = keyringBackend
+	}
+}
+
+// WithLaunchpadCLI provides the name of the CLI application to call Launchpad CLI commands
+func WithLaunchpadCLI(cliCmd string) Option {
+	return func(c *ChainCmd) {
+		c.cliCmd = cliCmd
+	}
+}
+
+// StartCommand returns the command to start the daemon of the chain
+func (c ChainCmd) StartCommand(options ...string) step.Option {
+	command := append([]string{
+		commandStart,
+	}, options...)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// InitCommand returns the command to initialize the chain
+func (c ChainCmd) InitCommand(moniker string) step.Option {
+	command := []string{
+		commandInit,
+		moniker,
+	}
+	command = c.attachChainID(command)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// AddKeyCommand returns the command to add a new key in the chain keyring
+func (c ChainCmd) AddKeyCommand(accountName string) step.Option {
+	command := []string{
+		commandKeys,
+		"add",
+		accountName,
+		optionOutput,
+		constJSON,
+	}
+	command = c.attachKeyringBackend(command)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// ImportKeyCommand returns the command to import a key into the chain keyring from a mnemonic
+func (c ChainCmd) ImportKeyCommand(accountName string) step.Option {
+	command := []string{
+		commandKeys,
+		"add",
+		accountName,
+		optionRecover,
+	}
+	command = c.attachKeyringBackend(command)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// ShowKeyAddressCommand returns the command to print the address of a key in the chain keyring
+func (c ChainCmd) ShowKeyAddressCommand(accountName string) step.Option {
+	command := []string{
+		commandKeys,
+		"show",
+		accountName,
+		optionAddress,
+	}
+	command = c.attachKeyringBackend(command)
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// AddGenesisAccountCommand returns the command to add a new account in the genesis file of the chain
+func (c ChainCmd) AddGenesisAccountCommand(address string, coins string) step.Option {
+	command := []string{
+		commandAddGenesisAccount,
+		address,
+		coins,
+	}
+	return step.Exec(c.appCmd, c.attachHome(command)...)
+}
+
+// GentxCommand returns the command to generate a gentx for the chain
+func (c ChainCmd) GentxCommand(
+	validatorName string,
```

Comment:
I think it would cleaner if we don't require optional flags as arguments. Instead, maybe we can use a struct to pass these or `...Option`s.